### PR TITLE
support Postgres versions 14 and 15

### LINKED ISSUE #9717: Postgres 15: permissions errors on public schema
Public schema permissions are much stricter in Postgres 15; we'll need to modify the installer and/or guides to accommodate this change.

I bumped FlyWay to 9.20.1 to explicitly support Postgres 15 for good measure, but:

```
Caused by: org.flywaydb.core.internal.sqlscript.FlywaySqlScriptException: Migration  failed
-----------------
SQL State  : 42501
Error Code : 0
Message    : ERROR: permission denied for schema public
  Position: 14
Location   :  ()
Line       : 1
Statement  : CREATE TABLE "public"."flyway_schema_history" (
    "installed_rank" INT NOT NULL,
    "version" VARCHAR(50),
    "description" VARCHAR(200) NOT NULL,
    "type" VARCHAR(20) NOT NULL,
    "script" VARCHAR(1000) NOT NULL,
    "checksum" INTEGER,
    "installed_by" VARCHAR(100) NOT NULL,
    "installed_on" TIMESTAMP NOT NULL DEFAULT now(),
    "execution_time" INTEGER NOT NULL,
    "success" BOOLEAN NOT NULL
)
Caused by: org.postgresql.util.PSQLException: ERROR: permission denied for schema public
  Position: 14

Caused by: org.flywaydb.core.internal.sqlscript.FlywaySqlScriptException: Migration  failed
-----------------
SQL State  : 42501
Error Code : 0
Message    : ERROR: permission denied for schema public
  Position: 14
Location   :  ()
Line       : 1
Statement  : CREATE TABLE "public"."flyway_schema_history" (
    "installed_rank" INT NOT NULL,
    "version" VARCHAR(50),
    "description" VARCHAR(200) NOT NULL,
    "type" VARCHAR(20) NOT NULL,
    "script" VARCHAR(1000) NOT NULL,
    "checksum" INTEGER,
    "installed_by" VARCHAR(100) NOT NULL,
    "installed_on" TIMESTAMP NOT NULL DEFAULT now(),
    "execution_time" INTEGER NOT NULL,
    "success" BOOLEAN NOT NULL
)
Caused by: org.postgresql.util.PSQLException: ERROR: permission denied for schema public
  Position: 14
```

To resolve, we probably want the installer (and Dataverse-Ansible) to:
`GRANT ALL ON SCHEMA public TO dvndb` unless we just want to grant `CREATE`?

### PR DESCRIPTION
**Pull request by donsizemore**
**What this PR does / why we need it**:

add flyway and installer support for Postgres versions 14 and 15

**Which issue(s) this PR closes**:

- Closes #9717 

**Special notes for your reviewer**:

None

**Suggestions on how to test this**:

Install Dataverse using installer and warfile built from branch, run E2E tests.

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

No

**Is there a release notes update needed for this change?**:

Perhaps, to explain the support (but not requirement) of the newer Postgres versions.

**Additional documentation**:

Added: https://dataverse-guide--9877.org.readthedocs.build/en/9877/installation/prerequisites.html#postgresql

### CODE REVIEW

File: scripts/installer/install.py [hunk 1]
```diff
@@ -380,12 +380,13 @@
...
+   # 3b. get the Postgres version for new permissions model in versions 15+
    try:
-      pg_full_version = conn.server_version
-      print("PostgresQL version: "+str(pg_full_version))
+      pg_full_version = str(conn.server_version)
+      pg_major_version = pg_full_version[0:2]
+      print("PostgreSQL version: "+pg_major_version)
    except:
-      print("Warning: Couldn't determine PostgresQL version.")
+      print("Warning: Couldn't determine PostgreSQL version.")
```
Comment (pdurbin) [inline scripts/installer/install.py:389]: 👍

**Review by pdurbin — APPROVED** [review]
I didn't test the installer but I tested in Docker. Looks good. Approved. Thanks, Don!

**Comment by pdurbin** [comment]
This is what I wrote [in Slack](https://iqss.slack.com/archives/C010LA04BCG/p1693597424258199?thread_ts=1693585451.435409&cid=C010LA04BCG) on Friday:

To summarize, @Don and I created https://github.com/gdcc/api-test-runner/actions/workflows/postgres15.yml and containerized Postgres 15 is working fine, (somewhat surprisingly) even without the "GRANT ALL ON SCHEMA public TO "+pgUser+" he added to install.py (our Docker images don’t use the installer).

The version being tested in that container is PostgreSQL 15.4 (Debian 15.4-1.pgdg120+1).

So yeah, I think we should be open to an issue and PR. His change to the installer is specific to PG 15:

    if int(pg_major_version) >= 15:
       conn_cmd = "GRANT ALL ON SCHEMA public TO "+pgUser+";"

Thanks or all the legwork, Don!

Also, I looked at the docs at https://guides.dataverse.org/en/5.14/installation/prerequisites.html and we already say, "We recommend installing the latest version that is available for your OS distribution." So this PR is in line with that. All the examples use Postgres 13 and we say, "The application has been tested with PostgreSQL versions up to 13." We can change the examples when we actually are ready to move developers, test infra, and prod to Postgres 13.

**Comment by kcondon** [comment]
Found an issue when installing on postgres 13:
Installer exits with: Is this correct? [y/n] y
performing database setup
Admin database connectivity succeeds.
PostgreSQL version: 13
Looks like the user already exists. Continuing.
Couldn't grant privileges on schema public to dvnapp

**Comment by pdurbin** [comment]
@donsizemore the fact that API tests are passing as of the second run is great: https://jenkins.dataverse.org/job/IQSS-Dataverse-Develop-PR/job/PR-9877/2/testReport/

My understanding is that PG 13 is being used for that test.

What about a test for PG 15? Would you be able to set up a dedicated Jenkins job for that? If we can get that to pass, I'm comfortable merging this.

**Comment by kcondon** [comment]
This looks good to me.

**Review by pdurbin — APPROVED** [review]
Looks good to me too.

**Pull request merged**